storage/file_persister.go

```diff
@@ -0,0 +1,14 @@
+package storage
+
+import (
+	"context"
+	"io"
+
+	"github.com/grafana/xk6-browser/log"
+)
+
+// FilePersister will persist files. It abstracts away the where and how of
+// writing files to the source destination.
+type FilePersister interface {
+	Persist(ctx context.Context, logger *log.Logger, path string, data io.Reader) error
```

Comment:
Why do we need an interface here? Since Go interfaces are explicit, we could declare it on the call side.

Besides, the signature is too fat. A value implementing this interface can use its field logger. The rest seems fine. Will we have a `path` on the cloud? Let's first define the concrete types and, finally, the interface.

Check [this](https://go.dev/wiki/CodeReviewComments#interfaces) for more information.

Reply:
Yep, I agree that this is the wrong way around in go. I set up the sub issues in such a way so that the local and remote persister could be done in parallel hence the initial PR to surface the interface that could be used by both. In hindsight that won't be how i do it next time since it's tricky to that without the concrete type.

> Besides, the signature is too fat. A value implementing this interface can use its field logger
I agree, and I don't like the addition of the `logger` as part of the method signiture. Take a look at this (unfinished) [PR](https://github.com/grafana/xk6-browser/pull/1197) on how it is instantiated and used. I don't think there's a way to instantiate it with a logger when the `ModuleInstance` is created. I could be wrong though.

> Will we have a path on the remote?
Yes, we will.

Reply:
The logger has been removed in de5f6bec23fe0c693db6f430981e2dd95a0d476a